tipg/middleware.py

```diff
@@ -51,19 +51,38 @@ async def dispatch(self, request: Request, call_next):
 class CatalogUpdateMiddleware(BaseHTTPMiddleware):
     """Middleware to update the catalog cache."""
 
+    def __init__(
+        self,
+        app: ASGIApp,
+        ttl: int = 300,
```

Comment:
we could maybe add a special case when `ttl=0` meaning that we don't want any refresh 🤷‍♂️

Reply:
that could be handled at the application level
```python
if settings.catalog_ttl:
    app.add_middleware(CatalogUpdateMiddleware, ttl=settings.catalog_ttl)